During bond_open(), the bonding driver always set the slave active flag to
true if the bond is not in active/backup mode.  Bonding driver should let
the aggregator selection logic set the active flag.

diff --git a/drivers/net/bonding/bond_main.c b/drivers/net/bonding/bond_main.c
index 3891c08..536d298 100644
--- a/drivers/net/bonding/bond_main.c
+++ b/drivers/net/bonding/bond_main.c
@@ -3648,7 +3648,7 @@ static int bond_open(struct net_device *bond_dev)
 			if ((bond->params.mode == BOND_MODE_ACTIVEBACKUP)
 				&& (slave != bond->curr_active_slave)) {
 				bond_set_slave_inactive_flags(slave);
-			} else {
+			} else if (bond->params.mode != BOND_MODE_8023AD) {
 				bond_set_slave_active_flags(slave);
 			}
 		}
